pkg/scheduler/framework/plugins/volumebinding/binder.go

```diff
@@ -573,16 +573,15 @@ func (b *volumeBinder) bindAPIUpdate(ctx context.Context, pod *v1.Pod, bindings
 	// Do the actual prebinding. Let the PV controller take care of the rest
 	// There is no API rollback if the actual binding fails
 	for _, binding = range bindings {
-		klog.V(5).InfoS("bindAPIUpdate: binding PV to PVC", "pod", klog.KObj(pod), "PV", klog.KObj(binding.pv), "PVC", klog.KObj(binding.pvc))
 		// TODO: does it hurt if we make an api call and nothing needs to be updated?
-		klog.V(2).InfoS("Claim bound to volume", "PVC", klog.KObj(binding.pvc), "PV", klog.KObj(binding.pv))
+		klog.V(2).InfoS("Updating PersistentVolume: binding to claim", "pod", klog.KObj(pod), "PV", klog.KObj(binding.pv), "PVC", klog.KObj(binding.pvc))
 		newPV, err := b.kubeClient.CoreV1().PersistentVolumes().Update(ctx, binding.pv, metav1.UpdateOptions{})
 		if err != nil {
-			klog.V(4).InfoS("Updating PersistentVolume: binding to claim failed", "PV", klog.KObj(binding.pv), "PVC", klog.KObj(binding.pvc), "err", err)
+			klog.V(4).InfoS("Updating PersistentVolume: binding to claim failed", "pod", klog.KObj(pod), "PV", klog.KObj(binding.pv), "PVC", klog.KObj(binding.pvc), "err", err)
 			return err
 		}
 
-		klog.V(4).InfoS("Updating PersistentVolume: bound to claim", "PV", klog.KObj(binding.pv), "PVC", klog.KObj(binding.pvc))
+		klog.V(2).InfoS("Updating PersistentVolume: bound to claim", "pod", klog.KObj(pod), "PV", klog.KObj(binding.pv), "PVC", klog.KObj(binding.pvc))
```

Comment:
@msau42 for my own understanding: what is the difference of the state of things at this point compared to line 747.

Is it worth logging both before and after calling `Update`? My gut feeling is that only after.

Reply:
There's a sequence of updates that happens on the PVC and PV objects to bind them together. The scheduler starts that sequence with this update, and then waits for the PV controller finishes the sequence. So L747 is when the PV controller has completed.

I think logging before could be a lower log level. The actual update will show up in api audit logs, and if there was an api server error, that would be returned here.

Reply:
Should I set the log level of L577 to V(4) (the log before performing the update) ?

Reply:
My main request is that the log line doesn't say "bound", as clearly the process didn't finish.

Logging in the scheduler makes it easy to debug, but we probably don't need both log lines. I also prefer only logging at V(2) after the Update call, but maybe we can be more explicit that there is more to come, as the process didn't finish at this point either, and it could be confusing to people oncall. What would be a good description @msau42?

Reply:
How about something like "Updated PersistentVolume with claim. Waiting for binding to complete"

Reply:
Thank you @msau42, done in d70858bcc6a

Reply:
It doesn't currently say: `Waiting for binding to complete`

Reply:
Oops, I was looking at the wrong line